Hashtable_.c

```diff
@@ -0,0 +1,197 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define CAPACITY 50000 // Size of the HashTable.
+
+unsigned long hash_function(char *str)
```

Comment:
if you are going to use a char * with unknown size i would recommend to write a struct with char * with his fixed size.